### PR TITLE
JGithub Improvements. Labels management

### PR DESCRIPTION
**Pull request by nprasath002**
Added some methods to manage labels in Github
- createLabel
- editLabel
- deleteLabel
- getLabel
- getLabels

### CODE REVIEW

**Comment by ianmacl** [comment]
Code looks reasonable.  Commits needs to be amended so that your proper email address is used (Github isn't picking up your identity correctly for the commits).  Also, tests need to be written for these added methods.

**Comment by realityking** [comment]
@nprasath002  Could you fix these as per Ian's comment. Would be great if you could rebase them into one commit as well.

File: libraries/joomla/github/http.php [hunk 1]
```diff
@@ -46,8 +46,9 @@ class JGithubHttp extends JHttp
 	 */
 	public function __construct(JRegistry $options = null, JHttpTransport $transport = null)
 	{
-		// Call the JHttp constructor to setup the object.
-		parent::__construct($options, $transport);
```
Comment (realityking) [inline libraries/joomla/github/http.php:50]: Why are you removing this? This increases the chance of breaking things when something is added to the JHttp constructor that is required.

File: libraries/joomla/github/issues.php [hunk 2]
```diff
@@ -107,6 +107,45 @@ public function createComment($user, $repo, $issueId, $body)
...
+   	 * Method to create a label on a repo.
+   	 *
+   	 * @param   string   $user     The name of the owner of the GitHub repository.
+   	 * @param   string   $repo     The name of the GitHub repository.
+   	 * @param   string   $name     The label name.
+   	 * @param   string   $color    The label color.
+   	 *
+   	 * @return  object
+   	 *
+   	 * @since   12.1
```
Comment (realityking) [inline libraries/joomla/github/issues.php:120]: Please change these to 12.2.

Comment (LouisLandry) [inline libraries/joomla/github/issues.php:120]: And by that he means 12.3 ;-)

**Comment by LouisLandry** [comment]
@nprasath002 would love to get this merged into 12.3 if you have the time for the little cleanup necessary.

**Comment by LouisLandry** [comment]
OK, @nprasath002 this has been marked for including in the 12.3 release (about 2 months out).  If you can find the time to make the adjustments listed above we'd love to merge it.  If you have questions about how to do any of it please feel free to either ask in this thread, or ask on the platform mailing list.

For now I'm going to close this pull request until you've got it cleaned up and ready for further review.  When you get that done please re-open it and we'll get it sorted out then.  Thanks a bunch for the work on this, it will prove very useful!

**Comment by nprasath002** [comment]
I cleanedup and made a new pull request.
Thanks!

**Pull request closed without merging**